GameEngine/StochasticDeck.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace GameEngine
+{
+    public class StochasticDeck : Deck
+    {
+        private Dictionary<CardType, int> CardsByCount;
+
+        public Dictionary<CardType, int> CardWeights
+        {
+            get { return new Dictionary<CardType, int>(CardsByCount); }
+        }
+
+        public override List<CardType> Cards { get => SampleAll().ToList(); }
+
+        private int RunningCount { get; set; }
+        public override int Count => RunningCount;
+
+        private StochasticDeck() { }
```

Comment:
💡 The existence of a public constructor that takes more than 1 argument means you do not need to explicitly define the private default constructor

Reply:
This constructor is used in the `Clone` method.